arcgis-ios-sdk-samples/Augmented reality/Navigate in AR/NavigateARNavigatorViewController.swift

```diff
@@ -0,0 +1,323 @@
+    ) {
+        setSpeakDirection(with: voiceGuidance.text)
+    }
+    
+    func routeTrackerRerouteDidStart(_ routeTracker: AGSRouteTracker) {
+        setStatus(message: "Rerouting...")
+    }
+    
+    func routeTracker(_ routeTracker: AGSRouteTracker, didUpdate trackingStatus: AGSTrackingStatus) {
+        switch trackingStatus.destinationStatus {
+        case .notReached, .approaching:
+            let currentManeuver = currentRoute.directionManeuvers[trackingStatus.currentManeuverIndex]
+            setStatus(message: currentManeuver.directionText)
+        case .reached:
+            setStatus(message: "You have arrived!")
+        default:
+            return
+        }
+    }
+    
+    func routeTracker(
+        _ routeTracker: AGSRouteTracker,
+        rerouteDidCompleteWith trackingStatus: AGSTrackingStatus?,
+        error: Error?
+    ) {
+        if let latestRoute = trackingStatus?.routeResult.routes.first {
+            if latestRoute != self.currentRoute {
+                setRouteAndGraphic(route: latestRoute)
+            }
+        }
+    }
+    
+    func setSpeakDirection(with text: String?) {
+        speechSynthesizer.stopSpeaking(at: AVSpeechBoundary.word)
+        if let text = text {
+            let speechUtterance = AVSpeechUtterance(string: text)
+            speechUtterance.rate = AVSpeechUtteranceMaximumSpeechRate * 0.5
+            speechSynthesizer.speak(speechUtterance)
+        }
+    }
+}
+
+// MARK: - AGSLocationChangeHandlerDelegate
+
+extension NavigateARNavigatorViewController: AGSLocationChangeHandlerDelegate {
+    func locationDataSource(
+        _ locationDataSource: AGSLocationDataSource,
+        locationDidChange location: AGSLocation
+    ) {
+        startButtonItem.isEnabled = true
+        // Send location updates to route tracker if actively navigating.
+        routeTracker?.trackLocation(location)
+    }
+}
+
+// MARK: - Calibration popup
+
+extension NavigateARNavigatorViewController {
+    @IBAction func showCalibrationPopup(_ sender: UIBarButtonItem) {
+        if isCalibrating {
+            calibrationVC.dismiss(animated: true)
+        } else {
+            showPopup(calibrationVC, sourceButton: sender)
+        }
+        isCalibrating.toggle()
+    }
+    
+    private func showPopup(_ controller: UIViewController, sourceButton: UIBarButtonItem) {
+        controller.modalPresentationStyle = .popover
+        if let presentationController = controller.popoverPresentationController {
+            presentationController.delegate = self
+            presentationController.barButtonItem = sourceButton
+            presentationController.permittedArrowDirections = [.down, .up]
+        }
+        present(controller, animated: true)
+    }
+}
+
+extension NavigateARNavigatorViewController: UIPopoverPresentationControllerDelegate {
+    func popoverPresentationControllerDidDismissPopover(
+        _ popoverPresentationController: UIPopoverPresentationController
+    ) {
```

Comment:
While this sort of thing (putting each parameter on its own line) does make for shorter lines, I'm not sure it is necessary. For one thing, none of the other samples do that, at least in general. That also isn't the way that Xcode autocompletes them. I recommend sticking to the existing convention of keeping function declaration parameters all together on one line.